Flaky-test: PartitionedSystemTopicTest.cleanup

<!--- 

Instructions for reporting a flaky test using this issue template:

1. Replace test_class in title and body with the short test class name WITHOUT the package name.
2. Replace test_method in title and body with the test method that failed. Multiple methods are flaky, remove the content that refers to the test method.
3. Replace "url here" with a url to an example failure. In the Github Actions workflow run logs, you can right click on the line number to copy a link to the line. Example of such url is https://github.com/apache/pulsar/pull/8892/checks?check_run_id=1531075794#step:9:377 . The logs are available for a limited amount of time (usually for a few weeks).
4. Replace "relevant parts of the exception stacktrace here" with the a few lines of the stack trace that shows at least the exception message and the line of test code where the stacktrace occurred.
5. Replace "full exception stacktrace here" with the full exception stacktrace from logs. This section will be hidden by default.
6. Remove all unused fields / content to unclutter the reported issue. Remove this comment too.

-->
PartitionedSystemTopicTest.cleanup is flaky. It fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/6721616096?check_suite_focus=true#step:10:1876)

```
  Error:  Tests run: 9, Failures: 1, Errors: 0, Skipped: 6, Time elapsed: 83.268 s <<< FAILURE! - in org.apache.pulsar.broker.systopic.PartitionedSystemTopicTest
  Error:  cleanup(org.apache.pulsar.broker.systopic.PartitionedSystemTopicTest)  Time elapsed: 71.628 s  <<< FAILURE!
  org.apache.pulsar.broker.PulsarServerException: org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.ExecutionException: org.apache.pulsar.common.util.FutureUtil$LowOverheadTimeoutException: Closing producers and consumers timed out.
  	at org.apache.pulsar.broker.PulsarService.closeAsync(PulsarService.java:574)
  	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
  	at org.apache.pulsar.broker.PulsarService.closeAsync(PulsarService.java:395)
  	at org.apache.pulsar.broker.PulsarService.close(PulsarService.java:374)
  	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
  	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
  Caused by: org.apache.pulsar.client.api.PulsarClientException: java.util.concurrent.ExecutionException: org.apache.pulsar.common.util.FutureUtil$LowOverheadTimeoutException: Closing producers and consumers timed out.
  	at org.apache.pulsar.client.api.PulsarClientException.unwrap(PulsarClientException.java:1074)
  	at org.apache.pulsar.client.impl.PulsarClientImpl.close(PulsarClientImpl.java:711)
  	at org.apache.pulsar.broker.PulsarService.closeAsync(PulsarService.java:491)
  	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
  	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor$Dispatcher$ByteBuddy$Kf8nzKi5.invokeWithArguments(Unknown Source)
  	at org.mockito.internal.util.reflection.InstrumentationMemberAccessor.invoke(InstrumentationMemberAccessor.java:239)
  	at org.mockito.internal.util.reflection.ModuleMemberAccessor.invoke(ModuleMemberAccessor.java:55)
  	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.tryInvoke(MockMethodAdvice.java:333)
  	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.access$500(MockMethodAdvice.java:60)
  	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice$RealMethodCall.invoke(MockMethodAdvice.java:253)
  	at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:142)
  	at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:45)
  	at org.mockito.Answers.answer(Answers.java:99)
  	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:110)
  	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
  	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:34)
  	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:82)
  	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.handle(MockMethodAdvice.java:151)
  	at org.apache.pulsar.broker.PulsarService.closeAsync(PulsarService.java:395)
  	at org.apache.pulsar.broker.PulsarService.close(PulsarService.java:374)
```

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java
@@ -750,24 +750,23 @@ public class PulsarClientImpl implements PulsarClient {
         FutureUtil.addTimeoutHandling(combinedFuture, Duration.ofSeconds(CLOSE_TIMEOUT_SECONDS),
                 shutdownExecutor, () -> FutureUtil.createTimeoutException("Closing producers and consumers timed out.",
                         PulsarClientImpl.class, "closeAsync"));
-        combinedFuture.whenComplete((__, t) -> new Thread(() -> {
+        combinedFuture.handle((__, t) -> {
             if (t != null) {
                 log.error("Closing producers and consumers failed. Continuing with shutdown.", t);
             }
-            shutdownExecutor.shutdownNow();
-            // All producers & consumers are now closed, we can stop the client safely
-            try {
-                shutdown();
-                closeFuture.complete(null);
+            new Thread(() -> {
+                shutdownExecutor.shutdownNow();
+                // All producers & consumers are now closed, we can stop the client safely
+                try {
+                    shutdown();
+                } catch (PulsarClientException e) {
+                    log.error("Shutdown failed. Ignoring the exception.", e);
+                }
                 state.set(State.Closed);
-            } catch (PulsarClientException e) {
-                closeFuture.completeExceptionally(e);
-            }
-        }, "pulsar-client-shutdown-thread").start()).exceptionally(exception -> {
-            closeFuture.completeExceptionally(exception);
+                closeFuture.complete(null);
+            }, "pulsar-client-shutdown-thread").start();
             return null;
         });
-
         return closeFuture;
     }
 

